### PR TITLE
Added properly units to rows count in public datasets view

### LINKED ISSUE #4037: Number of rows is not reduced in public list of dataset
In the public list of a dataset, you can find something like:

image

While in the Editor it would be collapsed to 9.0M 

@xavijam

### PR DESCRIPTION
**Pull request by xavijam**
Rows count with thousand delimiters + humanize format, like:

5,000 rows
180.32K
2.65M

cc @iriberri.

Fixes #4037.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4326/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4325/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4328/))

**Comment by xavijam** [comment]
@kartones, could you take a look please?

File: app/views/admin/pages/public_datasets.html.erb [hunk 1]
```diff
@@ -20,10 +20,17 @@
...
-                  <i class="iconFont iconFont-Rows RowsIndicator-icon"></i>
-                  <%= pluralize(vis[:rows_count], 'row') %>
-                </div>
+                <% if !vis[:rows_count].blank? %>
+                  <div class="RowsIndicator">
+                    <i class="iconFont iconFont-Rows RowsIndicator-icon"></i>
+                    <% if vis[:rows_count] < 10000 %>
+                      <%= number_with_delimiter(vis[:rows_count]) %>
+                    <% else %>
+                      <%= number_to_human(vis[:rows_count], units: { unit:"", thousand: "K", million: "M" }, precision: 2, format: '%n%u') %>
```
Comment (xavijam) [inline app/views/admin/pages/public_datasets.html.erb]: We are doing the same in the private dashboard, not formating the number with units if the size is smaller than 10,000.

**Comment by juanignaciosl** [comment]
LGTM

**Pull request merged**